### PR TITLE
updating to Terraform provider framework v1.0.1

### PR DESCRIPTION
**Pull request by Sakshi-dell**
# Description
Updating terraform provider for powerstore to framework v1.0.1

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Acceptance test ran successfully

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# These owners will be the default owners for everything in the repo
+
+* @Sakshi-dell @mkmayank-dell @prajwalpatil25 @shenda1
```
Comment (shenda1) [inline .github/CODEOWNERS]: remove @mkmayank-dell

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -1,36 +1,101 @@
...
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Testing only needs permissions to read the repository contents.
+permissions:
+  contents: read
+env:
+  # Go language version to use for building. This value should also be updated
+  # in the testing workflow if changed.
+  GO_VERSION: '1.18'
```
Comment (Krunal-Thakkar) [inline .github/workflows/tests.yml]: We are using Go 1.19, Correct?

File: go.mod [hunk 3]
```diff
@@ -1,11 +1,85 @@
...
-	github.com/golang/mock v1.4.4
-	github.com/hashicorp/terraform-plugin-sdk/v2 v2.0.4
-	github.com/jarcoal/httpmock v1.0.6
-	github.com/stretchr/testify v1.6.1
-	golang.org/x/tools v0.1.0 // indirect
+	github.com/dell/gopowerstore v1.9.1-0.20230112140829-870b2aa283a3
+	github.com/hashicorp/terraform-plugin-docs v0.13.0
+	github.com/hashicorp/terraform-plugin-framework v1.0.1
+	github.com/hashicorp/terraform-plugin-framework-validators v0.9.0
+	github.com/hashicorp/terraform-plugin-go v0.14.2
```
Comment (Krunal-Thakkar) [inline go.mod]: 0.14.3 is available 

Do we have any process to consider the updated version?

File: powerstore/resource_volume_test.go [hunk 4]
```diff
@@ -0,0 +1,852 @@
...
+		Steps: []resource.TestStep{
+			{
+				Config:      VolumeParamsWithInvalidCapUnit,
+				ExpectError: regexp.MustCompile("Invalid Attribute Value Match"),
+			},
+		},
+	})
+}
+
+// todo fix this test case, since it is not working with validation on capacity unit
```
Comment (Krunal-Thakkar) [inline powerstore/resource_volume_test.go]: TODO committed is it fine?

Comment (Sakshi-dell) [reply powerstore/resource_volume_test.go]: This is acceptance test and coverage is not impacted so can be taken up in next release.

**Review by Krunal-Thakkar — DISMISSED** [review]

File: docs/resources/protectionpolicy.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+page_title: "powerstore_protectionpolicy Resource - powerstore"
+subcategory: ""
+description: |-
+  ProtectionPolicy resource
+---
+
+# powerstore_protectionpolicy (Resource)
+
+ProtectionPolicy resource
+
```
Comment (prajwalpatil25) [inline docs/resources/protectionpolicy.md:12]: We'd need to add example for protection policy resource

**Review by shenda1 — DISMISSED** [review]

**Review by shenda1 — APPROVED** [review]

**Review by Krunal-Thakkar — APPROVED** [review]

**Pull request merged**